encord/objects/ontology_labels_impl.py

```diff
@@ -1056,26 +1078,99 @@ def initialise_labels(
         """
         if self.is_labelling_initialised and not overwrite:
             raise LabelRowError(
                 "You are trying to re-initialise a label row that has already been initialised. This would overwrite "
                 "current labels. If this is your intend, set the `overwrite` flag to `True`."
             )
 
-        get_signed_url = False
-        if self.label_hash is None:
-            label_row_dict = self._project_client.create_label_row(self.data_hash)
-        else:
-            label_row_dict = self._project_client.get_label_row(
+        if bundle is not None:
+            self.__batched_initialise(
+                bundle,
                 uid=self.label_hash,
-                get_signed_url=get_signed_url,
+                get_signed_url=False,
                 include_object_feature_hashes=include_object_feature_hashes,
                 include_classification_feature_hashes=include_classification_feature_hashes,
                 include_reviews=include_reviews,
             )
+        else:
+            if self.label_hash is None:
+                label_row_dict = self._project_client.create_label_row(self.data_hash)
+            else:
+                label_row_dict = self._project_client.get_label_row(
+                    uid=self.label_hash,
+                    get_signed_url=False,
+                    include_object_feature_hashes=include_object_feature_hashes,
+                    include_classification_feature_hashes=include_classification_feature_hashes,
+                    include_reviews=include_reviews,
+                )
 
-        self.from_labels_dict(label_row_dict)
+            self.from_labels_dict(label_row_dict)
+
+    def __batched_initialise(
+        self,
+        batch: Bundle,
+        uid,
+        get_signed_url,
+        include_object_feature_hashes,
+        include_classification_feature_hashes,
+        include_reviews,
+    ):
+        if self.label_hash is None:
+            batch.add(
+                operation=self._project_client.create_label_rows,
+                request_reducer=self._bundle_create_rows_reducer,
+                result_mapper=BundleResultMapper(
+                    mapping_function=self._bundle_create_rows_mapper,
+                    result_handler=BundleResultHandler(predicate=self.data_hash, handler=self.from_labels_dict),
+                ),
+                payload=BundledCreateRowsPayload(
+                    uids=[self.data_hash],
+                ),
+                limit=LABEL_ROW_BUNDLE_CREATE_LIMIT,
+            )
+        else:
+            batch.add(
+                operation=self._project_client.get_label_rows,
+                request_reducer=self._bundle_get_rows_reducer,
+                result_mapper=BundleResultMapper(
+                    mapping_function=self._bundle_get_rows_mapper,
+                    result_handler=BundleResultHandler(predicate=self.label_hash, handler=self.from_labels_dict),
```

Comment:
I understand what is going on here, but I think this part confused me the most. Maybe I find it unusual that we have the `predicate` and the `mapping_function` in two different places and I'd have expected them to be one place. But maybe it's also just hard to simplify this problem.